### DISCUSSION TITLE
Cannot access host FS from Linux guest

### DISCUSSION THREAD

**Discussion by freebrowser1** [utmapp/UTM#3786] [UTM Help (macOS)]
**Describe the issue**
From an Ubuntu guest (with Spice tools installed by `apt install spice-vdagent spice-webdavd davfs2`) I could not get access to the host.
I tried mounting via WebDAV by adding it to `/etc/fstab` in Linux which worked after a lot of hassle, but after remounting the folder in the mount point is owned by root, so the host shared folder cannot be accessed by a regular user.
This is not particularly a UTM issue, but the issue is actually the inability to access the host shared folder.
Is there a simpler way to access the host shared folder  ?

EDIT: Later on, I found a webpage which tells something about it. 
https://support.cci.drexel.edu/software-courses/utm-virtualization-m1-macs/utm-sharing-folder-linux-guest/
The UTM documentation does not tell anything about this possibility.

It tells that there is a folder in a Linux file browser but it does not always work. After changing the shared folder it did not work anymore.

**Configuration**
* UTM Version: 3.1.5
* OS Version:  macOS Monterey 10.17.2
* Intel or Apple Silicon? Apple Silicon for both Host and Linux Guest

**Upload VM**
````<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>ConfigurationVersion</key>
	<integer>2</integer>
	<key>Debug</key>
	<dict/>
	<key>Display</key>
	<dict>
		<key>ConsoleBackgroundColor</key>
		<string>#000000</string>
		<key>ConsoleFont</key>
		<string>Menlo-Regular</string>
		<key>ConsoleFontSize</key>
		<integer>12</integer>
		<key>ConsoleOnly</key>
		<false/>
		<key>ConsoleTextColor</key>
		<string>#ffffff</string>
		<key>ConsoleTheme</key>
		<string>Default</string>
		<key>DisplayCard</key>
		<string>virtio-ramfb</string>
		<key>DisplayDownscaler</key>
		<string>linear</string>
		<key>DisplayUpscaler</key>
		<string>nearest</string>
	</dict>
	<key>Drives</key>
	<array>
		<dict>
			<key>DriveName</key>
			<string>drive1</string>
			<key>ImageType</key>
			<string>cd</string>
			<key>InterfaceType</key>
			<string>usb</string>
			<key>Removable</key>
			<true/>
		</dict>
		<dict>
			<key>DriveName</key>
			<string>drive0</string>
			<key>ImagePath</key>
			<string>data.qcow2</string>
			<key>ImageType</key>
			<string>disk</string>
			<key>InterfaceType</key>
			<string>virtio</string>
		</dict>
		<dict>
			<key>DriveName</key>
			<string>drive9141C451-56CD-4C70-8F74-00494D386199</string>
			<key>ImagePath</key>
			<string>efi_vars.fd</string>
			<key>ImageType</key>
			<string>none</string>
			<key>InterfaceType</key>
			<string></string>
		</dict>
	</array>
	<key>Info</key>
	<dict>
		<key>Icon</key>
		<string>linux</string>
	</dict>
	<key>Input</key>
	<dict>
		<key>InputLegacy</key>
		<false/>
	</dict>
	<key>Networking</key>
	<dict>
		<key>NetworkCard</key>
		<string>virtio-net-pci</string>
		<key>NetworkCardMAC</key>
		<string>D2:EB:A1:52:84:58</string>
		<key>NetworkMode</key>
		<string>bridged</string>
	</dict>
	<key>Printing</key>
	<dict/>
	<key>Sharing</key>
	<dict>
		<key>ClipboardSharing</key>
		<true/>
		<key>DirectoryReadOnly</key>
		<false/>
		<key>DirectorySharing</key>
		<true/>
		<key>Usb3Support</key>
		<false/>
		<key>UsbRedirectMax</key>
		<integer>3</integer>
	</dict>
	<key>Sound</key>
	<dict>
		<key>SoundCard</key>
		<string>intel-hda</string>
		<key>SoundEnabled</key>
		<true/>
	</dict>
	<key>System</key>
	<dict>
		<key>Architecture</key>
		<string>aarch64</string>
		<key>BootDevice</key>
		<string></string>
		<key>BootUefi</key>
		<true/>
		<key>CPU</key>
		<string>default</string>
		<key>CPUCount</key>
		<integer>0</integer>
		<key>ForcePS2Controller</key>
		<false/>
		<key>Memory</key>
		<integer>2048</integer>
		<key>RTCUseLocalTime</key>
		<true/>
		<key>RngEnabled</key>
		<true/>
		<key>SystemUUID</key>
		<string>0F5C28FB-DE75-442A-9F62-B6F46A26A61F</string>
		<key>Target</key>
		<string>virt</string>
		<key>UseHypervisor</key>
		<true/>
	</dict>
</dict>
</plist>
````

**Comment by osy** [comment]
If you're using Ubuntu, it should have installed a systemd service. You can then create your own .service and depends on it and mounts to the right point on startup. I think if you download the prebuilt Debian Xfce image from the gallery, you can see an example of how to do that (Debian also uses systemd).

**Reply by freebrowser1** [reply]
After installing xfce and I found a mount folder `/run/user/1000/gvfs/` with all files and folders under it which were the shared folder of my Macbook.
But later on, after a reboot, the folders were gone, the aforementioned folder was gone. I did not remove anything, so what went wrong ?

**Comment by freebrowser1** [comment]
<img width="1328" alt="Screenshot 2022-03-17 at 21 00 20" src="https://user-images.githubusercontent.com/26625141/158885811-9ef66a35-795a-4be0-8d36-141439e76bd6.png">

**Comment by fiore42** [comment]
None of these suggestions worked for me.

I finally found joy following [these steps](https://docs.getutm.app/guest-support/linux/#virtfs).

Host: MacOS Ventura 13.0 on MacBook Air M2
Guest: Ubuntu 22.04 for ARM
UTM 4.0.9

hope it helps!

**Comment by martin-braun** [comment]
The `VirtFS` method caused some headache for me in regards of permissions, so I attempted to get this to work in Ubuntu Server and it worked, but auto-mounting at boot does not work. `mount -a` will mount it for me. But it's terribly slow. 

Here is what I got so far: https://github.com/utmapp/docs.getutm.app/pull/21

**Comment by freebrowser1** [comment]
I found this solution, I use `sshfs` on the guest side and use `/media/sf_user/` as an example folder to mount to, but any other folder can be used, as long the user named `user` here has read and write access to that folder. Furthermore use 'Emulated VLAN' as network type, which can be used alongside 'Bridged' as well, but the latter is not required.
Here I use an SSH tunnel to the host if you want to use port forwarding to the client with virtual ports for SSH, Cockpit and http to the guest, but that is not mandatory.
The idea is that the host has a fixed internal IP address, rather than a network dependant one using Bridged.
I use this successfully on Ubuntu 22.04 LTS guest on a macOS ventura host and works in any wifi network.

```                                      
#!/bin/sh
IP="10.0.2.2"
if [ ! -d "/media/sf_user/Documents" ]; then
   sshfs -o allow_other,uid=1000,gid=33,ConnectTimeout=6  user@$IP:/Users/user/ /media/sf_user
   echo "Mount done !"
   # Use SSH tunnel (not mandatory for mounting)
   ssh -R 2023:localhost:22 -R 2290:localhost:9090 -R 2259:localhost:5901 -R 2243:localhost:443 -R 2280:localhost:80 -N user@10.0.2.2 &
   echo "Port forward done !"
fi
```
